slick/src/sphinx/developers.rst

```diff
@@ -0,0 +1,155 @@
+capabilities which are not actually implemented. For this purpose the Slick
+unit tests have been factored out into a separate Slick TestKit project.
+
+To get started, you can clone the `Slick TestKit Example`_ project which
+contains a (slightly outdated) version of Slick's standard PostgreSQL driver
+and all the infrastructure required to build and test it.
+
+Scaffolding
+___________
+
+Its ``build.sbt`` file is straight-forward. Apart from the usual name and
+version settings, it adds the dependencies for Slick, the TestKit,
+junit-interface, Logback and the PostgreSQL JDBC driver, and it sets some
+options for the test runs:
+
+.. parsed-literal::
+  libraryDependencies ++= Seq(
+    "com.typesafe.slick" %% "slick" % "|release|",
+    "com.typesafe.slick" %% "slick-testkit" % "|release|" % "test",
+    "com.novocode" % "junit-interface" % "0.10" % "test",
+    "ch.qos.logback" % "logback-classic" % "0.9.28" % "test",
+    "postgresql" % "postgresql" % "9.1-901.jdbc4" % "test"
+  )
+
+  testOptions += Tests.Argument(TestFrameworks.JUnit, "-q", "-v", "-s", "-a")
+
+  parallelExecution in Test := false
+
+  logBuffered := false
+
+There is a copy of Slick's logback configuration in
+``src/test/resources/logback-test.xml`` but you can swap out the logging
+framework if you prefer a different one.
+
+Driver
+______
+
+The actual driver implementation can be found under ``src/main/scala``.
+
+Test Harness
+____________
+
+In order to run the TestKit tests, you need to add a class that extends
+``DriverTest``, plus an implementation of ``TestDB`` which tells the TestKit
+how to connect to a test database, get a list of tables, clean up between
+tests, etc.
+
+In the case of the PostgreSQL test harness (in
+``src/test/scala/scala/slick/driver/test/MyPostgresTest.scala``) most of the
+default implementations can be used out of the box::
+
+  @RunWith(classOf[Testkit])
+  class MyPostgresTest extends DriverTest(MyPostgresTest.tdb)
+
+  object MyPostgresTest {
+    def tdb(cname: String) = new ExternalTestDB("mypostgres", MyPostgresDriver) {
+      override def getLocalTables(implicit session: Session) = {
+        val tables = ResultSetInvoker[(String,String,String, String)](_.conn.getMetaData()
+                      .getTables("", "public", null, null))
+        tables.list.filter(_._4.toUpperCase == "TABLE").map(_._3).sorted
+      }
+      override def getLocalSequences(implicit session: Session) = {
+        val tables = ResultSetInvoker[(String,String,String, String)](_.conn.getMetaData()
+                      .getTables("", "public", null, null))
+        tables.list.filter(_._4.toUpperCase == "SEQUENCE").map(_._3).sorted
+      }
+      override lazy val capabilities = driver.capabilities + TestDB.plainSql
+    }
+  }
+
+Database Configuration
+______________________
+
+Since the PostgreSQL test harness is based on ``ExternalTestDB``, it needs to
+be configured in ``test-dbs/databases.properties``::
+
+  # PostgreSQL quick setup:
```

Comment:
I had a warning on this line (135):

```
developers.rst:135: WARNING: Could not parse literal_block as "scala". highlighting skipped.
```

...but I do for other files in the documentation. So not huge.

...time passes...

OK, I've done some digging and it looks like this will make the error go away:

![ini](https://cloud.githubusercontent.com/assets/102661/13680645/767d0452-e6f1-11e5-9d37-cc94a692935a.png)

Notice: one colon at the end of line 133 for this to work.